monkey/infection_monkey/network_scanning/scan_target_generator.py

```diff
@@ -58,35 +59,38 @@ def _range_to_addresses(range_obj: NetworkRange) -> List[NetworkAddress]:
     return addresses
 
 
-def _get_ips_from_ranges_to_scan(ranges_to_scan: List[str]) -> List[NetworkAddress]:
-    scan_targets = []
-
+def _get_ips_from_subnets_to_scan(subnets_to_scan: List[str]) -> List[NetworkAddress]:
     ranges_to_scan = NetworkRange.filter_invalid_ranges(
-        ranges_to_scan, "Bad network range input for targets to scan:"
+        subnets_to_scan, "Bad network range input for targets to scan:"
     )
 
     network_ranges = [NetworkRange.get_range_obj(_range) for _range in ranges_to_scan]
+    return _get_ips_from_ranges_to_scan(network_ranges)
+
+
+def _get_ips_from_ranges_to_scan(network_ranges: List[NetworkRange]) -> List[NetworkAddress]:
+    scan_targets = []
 
     for _range in network_ranges:
         scan_targets.extend(_range_to_addresses(_range))
     return scan_targets
 
 
 def _get_ips_to_scan_from_local_interface(
-    interfaces: List[NetworkInterface],
+    interfaces: List[IPv4Interface],
 ) -> List[NetworkAddress]:
-    ranges = [f"{interface.address}{interface.netmask}" for interface in interfaces]
+    ranges = [f"{str(interface.ip)}/{interface.network.prefixlen}" for interface in interfaces]
```

Comment:
What about

```suggestion
    ranges = map(str, interfaces)
```
`str()` should work:
```
>>> i = IPv4Interface("192.168.1.10/24")
>>> str(i)
'192.168.1.10/24'
>>> 
```